etc/config/assembly.amazon.properties

```diff
@@ -8,82 +8,88 @@ defaultCompiler=nasm21303
 group.nasm.compilers=nasm21202:nasm21302:nasm21303
 group.nasm.versionFlag=-v
 group.nasm.options=-f elf -g -F stabs
-compiler.nasm21202.name=NASM 2.12.02
+group.nasm.isSemVer=true
+group.nasm.baseName=NASM
+compiler.nasm21202.semver=2.12.02
 compiler.nasm21202.exe=/opt/compiler-explorer/nasm-2.12.02/nasm
-compiler.nasm21302.name=NASM 2.13.02
+compiler.nasm21302.semver=2.13.02
 compiler.nasm21302.exe=/opt/compiler-explorer/nasm-2.13.02/nasm
-compiler.nasm21303.name=NASM 2.13.03
+compiler.nasm21303.semver=2.13.03
 compiler.nasm21303.exe=/opt/compiler-explorer/nasm-2.13.03/nasm
 
 group.gnuas.compilers=gnuas510:gnuas520:gnuas530:gnuas540:gnuas6:gnuas62:gnuas63:gnuas71:gnuas72:gnuas73:gnuas81:gnuassnapshot
 group.gnuas.versionFlag=--version
 group.gnuas.options=-g
+group.gnuas.isSemVer=true
+group.gnuas.baseName=x86-64 gcc
 compiler.gnuas510.exe=/opt/compiler-explorer/gcc-5.1.0/bin/as
-compiler.gnuas510.name=x86-64 gcc 5.1
+compiler.gnuas510.semver=5.1
 compiler.gnuas520.exe=/opt/compiler-explorer/gcc-5.2.0/bin/as
-compiler.gnuas520.name=x86-64 gcc 5.2
+compiler.gnuas520.semver=5.2
 compiler.gnuas530.exe=/opt/compiler-explorer/gcc-5.3.0/bin/as
-compiler.gnuas530.name=x86-64 gcc 5.3
+compiler.gnuas530.semver=5.3
 compiler.gnuas540.exe=/opt/compiler-explorer/gcc-5.4.0/bin/as
-compiler.gnuas540.name=x86-64 gcc 5.4
+compiler.gnuas540.semver=5.4
 compiler.gnuas6.exe=/opt/compiler-explorer/gcc-6.1.0/bin/as
-compiler.gnuas6.name=x86-64 gcc 6.1
+compiler.gnuas6.semver=6.1
 compiler.gnuas62.exe=/opt/compiler-explorer/gcc-6.2.0/bin/as
-compiler.gnuas62.name=x86-64 gcc 6.2
+compiler.gnuas62.semver=6.2
 compiler.gnuas63.exe=/opt/compiler-explorer/gcc-6.3.0/bin/as
-compiler.gnuas63.name=x86-64 gcc 6.3
+compiler.gnuas63.semver=6.3
 compiler.gnuas71.exe=/opt/compiler-explorer/gcc-7.1.0/bin/as
-compiler.gnuas71.name=x86-64 gcc 7.1
+compiler.gnuas71.semver=7.1
 compiler.gnuas72.exe=/opt/compiler-explorer/gcc-7.2.0/bin/as
-compiler.gnuas72.name=x86-64 gcc 7.2
+compiler.gnuas72.semver=7.2
 compiler.gnuas73.exe=/opt/compiler-explorer/gcc-7.3.0/bin/as
-compiler.gnuas73.name=x86-64 gcc 7.3
+compiler.gnuas73.semver=7.3
 compiler.gnuas81.exe=/opt/compiler-explorer/gcc-8.1.0/bin/as
-compiler.gnuas81.name=x86-64 gcc 8.1
+compiler.gnuas81.semver=8.1
 compiler.gnuassnapshot.exe=/opt/compiler-explorer/gcc-snapshot/bin/as
-compiler.gnuassnapshot.name=x86-64 gcc (trunk)
+compiler.gnuassnapshot.semver=(trunk)
 
 group.llvmas.compilers=llvmas30:llvmas31:llvmas32:llvmas33:llvmas341:llvmas350:llvmas351:llvmas37x:llvmas36x:llvmas371:llvmas380:llvmas381:llvmas390:llvmas391:llvmas400:llvmas401:llvmas500:llvmas600:llvmas_trunk
 group.llvmas.versionFlag=--version
 group.llvmas.options=-filetype=obj -o example.o
 group.llvmas.versionRe=LLVM version .*
+group.llvmas.isSemVer=true
+group.llvmas.baseName=x86-64 clang
 compiler.llvmas30.exe=/opt/compiler-explorer/clang+llvm-3.0-x86_64-linux-Ubuntu-11_10/bin/llvm-mc
-compiler.llvmas30.name=x86-64 clang 3.0
+compiler.llvmas30.semver=3.0.0
```

Comment:
Note `3.0.0` here and in C/C++ files.
This is to force the properties system to pick this up as a string, and not a number. A number here would change its value to "3" only, which would order it quite high